Core/src/IfcGeometryConverter/SplineConverter.h

```diff
@@ -211,28 +208,33 @@ namespace OpenInfraPlatform
 						carve::geom::vector<3> derivativeOne;
 						carve::geom::vector<3> derivativeTwo;
 
+						double length_i;
+						double curvature_i;
+
 						for (size_t i = 0; i < numCurvePoints; ++i) {
 							if (i == numCurvePoints - 1) { t = knotEnd - accuracy; }
 
 							curvePoint = computePointOfBSpline(order, t, controlPoints, numControlPoints, knotArray);
 							if (i == 0)
-								length[0] = 0;
+								length_i = 0;
 							else
-								length[i] = length[i-1] + (curvePoint - curvePointPrevious).length();
+								length_i = curvature[i-1].first + (curvePoint - curvePointPrevious).length();

```

Comment:
A more concise `C`-style-like expression would be:

    length_i = (i == 0 ? 0 : curvature[i] ... );

Reply:
Done in commit b7aa29d.
Because length_i is reduced to one line / statement, change of commit 9860558 is a suggestion from me; however, I'm not sure, whether it might got to complicated to read and understand.

Reply:
With the addition of the comments, it is transparent what is happening in the lines.